tests/setup_/_pattoo/test_systemd.py

```diff
@@ -0,0 +1,267 @@
+            file_path = os.path.join(temp_dir, 'test_file.txt')
+
+            # Create file
+            with open(file_path, 'w+') as test_file:
+                test_file.write('this is a test')
+            expected = [file_path]
+            result = _filepaths(temp_dir)
+            self.assertEqual(expected, result)
+
+    def test__copy_service_files(self):
+        """Testing method or function named "_copy_service_files"."""
+        # Initialize key variables
+        service_directory = '{1}/{0}systemd/system'.format('setup/', ROOT_DIR)
+        files = os.listdir(service_directory)
+
+        # Open agentd service file for reading
+        with open(os.path.join(service_directory, files[0]), 'r') as agentd:
+            pattoo_api_agentd = agentd.readlines()
+
+        # Open ingesterd service file for reading
+        with open(os.path.join(service_directory, files[1]), 'r') as ingesterd:
+            pattoo_ingesterd = ingesterd.readlines()
+
+        # Open apid service file for reading
+        with open(os.path.join(service_directory, files[2]), 'r') as apid:
+            pattoo_apid = apid.readlines()
+
+        # Put all service file contents into a list
+        expected_contents = sorted([
+                pattoo_api_agentd, pattoo_ingesterd, pattoo_apid])
+
+        # Initialize temporary directory to copy service files
+        with tempfile.TemporaryDirectory() as temp_dir:
+
+            # Initialize as set to ensure equality
+            expected = set([
+                os.path.join(temp_dir, 'pattoo_api_agentd.service'),
+                os.path.join(temp_dir, 'pattoo_apid.service'),
+                os.path.join(temp_dir, 'pattoo_ingesterd.service')
+            ])
+
+            copied_service_files = _copy_service_files(temp_dir)
+
+            # Ensure service files are copied to directory
+            result = set(copied_service_files)
+            self.assertEqual(result, expected)
+
+            # Ensure file contents are preserved
+            with open(os.path.join(
+                    service_directory,
+                    copied_service_files[0]), 'r') as agentd:
+                temp_api_agentd = agentd.readlines()
+
+            with open(os.path.join(
+                    service_directory,
+                    copied_service_files[1]), 'r') as ingesterd:
+                temp_ingesterd = ingesterd.readlines()
+
+            with open(os.path.join(
+                    service_directory,
+                    copied_service_files[2]), 'r') as apid:
+                temp_apid = apid.readlines()
+
+            actual_contents = sorted(
+                [temp_api_agentd, temp_ingesterd, temp_apid])
+
+            # Ensure contents are equal
+            self.assertEqual(expected_contents, actual_contents)
+
+    def test__symlink_dir(self):
+        """Testing method or function named "_symlink_dir"."""
+        # Initialise key variables
+        etc_dir = '/etc/systemd/system/multi-user.target.wants'
```

Comment:
This location depends on the operating system. In CentOS it is '/usr/lib/systemd/' You need to test for this in ALL cases including installation. The OS will not always be Ubuntu